storage/src/journal/benches/fixed_replay.rs

```diff
@@ -0,0 +1,82 @@
+use super::append_random_data;
+use commonware_runtime::{
+    benchmarks::{context, tokio},
+    tokio::Context,
+};
+use commonware_storage::journal::fixed::Journal;
+use commonware_utils::array::FixedBytes;
+use criterion::{black_box, criterion_group, Criterion};
+use futures::{pin_mut, StreamExt};
+use std::time::{Duration, Instant};
+
+/// Partition name to use in the journal config.
+const PARTITION: &str = "test_partition";
+
+/// Value of items_per_blob to use in the journal config.
+const ITEMS_PER_BLOB: u64 = 100_000;
+
+/// Size of each journal item in bytes.
+const ITEM_SIZE: usize = 32;
+
+/// Replay all items in the given `journal`.
+async fn bench_run(journal: &mut Journal<Context, FixedBytes<ITEM_SIZE>>, items_to_read: u64) {
+    let concurrency = (items_to_read / ITEMS_PER_BLOB) as usize;
+    let stream = journal
+        .replay(concurrency)
+        .await
+        .expect("failed to replay journal");
+    pin_mut!(stream);
+    while let Some(result) = stream.next().await {
+        match result {
+            Ok(item) => {
+                black_box(item);
+            }
+            Err(err) => panic!("Failed to read item: {}", err),
+        }
+    }
+}
+
+/// Benchmark the replaying of ITEMS_TO_WRITE (and then ITEMS_TO_WRITE*2)
+/// items from a journal containing exactly that number of items.
+fn bench_fixed_replay(c: &mut Criterion) {
+    let executor = tokio::Executor::default();
+    for items in [1_000_000, 5_000_000, 10_000_000] {
```

Comment:
this should have the same number of iterations as the fixed_read_sequential test so we can A/B how much speedup replay is yielding.

Reply:
(that or combine these tests with different "modes"?)